File: Python/stack.py
from linked_list import *
class Stack:
    lst = LinkedList()
    def __init__(self,elem=None):
        if elem:
            self.push(elem)
    def __len__(self):
        return len(self.lst)
    def isempty(self):
        return len(self.lst)==0
    def push(self,elem):
        self.lst.addfirst(elem)
    def pop(self):
        return self.lst.removefirst()
    def peek(self,index=0):
        return self.lst[index]
    def __getitem__(self,index):
        return self.peek(index)
    def __iter__(self):
        return self
    def __next__(self):
        return self.lst.__next__()
    def __repr__(self):
        return repr(self.lst)

File: DS_Implementations/Java/Stack.java
public class Stack<T> implements Iterable<T>{
    DoublyLinkedList<T> l = new DoublyLinkedList<T>();
    public Stack(){
        //pass;
    }
    public Stack(T elem){
        push(elem);
    }
    public void push(T elem){
        if(elem == null){
            throw new RuntimeException("Null elements are not supported");
        }
        l.addFirst(elem);
    }
    public void pop(){
        if(l.isEmpty()){
            throw new java.util.EmptyStackException();
        }
        l.removeFirst();
    }
    public T peek(){
        if(l.isEmpty()){
            throw new java.util.EmptyStackException();
        }
        return l.peekFirst();
    }
    public int size(){
        return l.size();
    }
    public boolean isEmpty(){
        return size() == 0;
    }

    @Override public java.util.Iterator<T> iterator(){
        return l.iterator();
    }
}